From 5d95b4adc861160c0a847d6c1963ebf8e1dbc59c Mon Sep 17 00:00:00 2001
From: Andrew Kelley <andrew@ziglang.org>
Date: Sun, 7 Sep 2025 20:23:19 -0700
Subject: [PATCH 099/133] std.mem.indexOfSentinel: eliminate unnecessary
 `@ptrCast`
X-Developer-Signature: v=1; a=openpgp-sha256; l=1107; i=xtex@astrafall.org;
 h=from:subject; bh=YgadulLoVWDNOp80r34DIAqlBnJde7UQ0f70/zUrnng=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78rjaJtz7YVt/M3vl0S6HyFsndZ/fscrVE7vSYbnss
 rvfoxM7OiayMIhxMViKKbIUGTZ4s+qk84suK5eFmcPKBDJEWqSBgYGBgYWBLzcxr9RIx0jPVNtQ
 z9BQx0jHiIGLUwCmOq+T4b/LnyvpR2/+v+EQlF34dvfBDRM5eAwXrQj+yWv6bMXst1NzGP4ps2f
 Ym01c0L8tId3Q7Hu8+MNth70iKk9VXvyxiafBwJ4HAA==
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

it was always unnecessary but now it's illegal

(cherry picked from commit dca4c302dd323f093c53b354ad64d504ed7a2dea)
---
 lib/std/mem.zig | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/std/mem.zig b/lib/std/mem.zig
index cc639500072e..1a67888c63bf 100644
--- a/lib/std/mem.zig
+++ b/lib/std/mem.zig
@@ -1166,10 +1166,10 @@ pub fn indexOfSentinel(comptime T: type, comptime sentinel: T, p: [*:sentinel]co
                     }
                 }
 
-                assert(std.mem.isAligned(@intFromPtr(&p[i]), block_size));
+                std.debug.assertAligned(&p[i], .fromByteUnits(block_size));
                 while (true) {
-                    const block: *const Block = @ptrCast(@alignCast(p[i..][0..block_len]));
-                    const matches = block.* == mask;
+                    const block: Block = p[i..][0..block_len].*;
+                    const matches = block == mask;
                     if (@reduce(.Or, matches)) {
                         return i + std.simd.firstTrue(matches).?;
                     }
-- 
2.52.0

